src/viewport/interface/viewport/Viewport.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstdint>
+
+class Viewport final
+{
+public:
+
+    Viewport(uint16_t width, uint16_t height)
+        : _width(width)
+        , _height(height)
+        , _aspect(static_cast<float>(_width) / _height)
+    { }
+
+    float get_aspect() const { return _aspect; }
+
+    uint16_t get_width() const { return _width; }
+
+    uint16_t get_height() const { return _height; }
+
+private:
+
+    uint16_t _width = 0;
+
+    uint16_t _height = 0;
+
+    float _aspect = 0;
+
```

Comment:
I see you like newlines very much

Reply:
Add const qualifier to all those fields as they are value initialized in the sole constructor.